assets/monitoring/grafana/v1alpha1/registry.go

```diff
@@ -52,3 +64,58 @@ var (
 	grafanaIngressTemplateString string
 	GrafanaIngressTemplate       = ParseObjectTemplateOrDie[*networkingv1.Ingress]("grafana-ingress", grafanaIngressTemplateString)
 )
+
+func parseDashboardsFromFS(files embed.FS, root string) (map[string]string, error) {
+	res := map[string]string{}
+
+	err := fs.WalkDir(files, root, func(p string, de fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if p == root || de.IsDir() {
+			return nil
+		}
+
+		data, err := fs.ReadFile(files, p)
+		if err != nil {
+			return fmt.Errorf("can't read file %q: %w", p, err)
+		}
+		relPath, err := filepath.Rel(root, p)
+		if err != nil {
+			return fmt.Errorf("can't compute relative path to file %q: %w", p, err)
+		}
+		res[relPath] = string(data)
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("can't walk fs: %w", err)
+	}
+
+	return res, nil
+}
+
+func gzipMapData(uncompressedMap map[string]string) (map[string]string, error) {
+	res := make(map[string]string, len(uncompressedMap))
+	for k, v := range uncompressedMap {
+		var buf bytes.Buffer
+		zw, err := gzip.NewWriterLevel(&buf, gzip.BestCompression)
```

Comment:
we could save on memory by directly writing into base64 encoder
```suggestion
		zw, err := gzip.NewWriterLevel(base64.NewEncoder(base64.StdEncoding, &buf),  gzip.BestCompression)
```

Reply:
done (but it needed a close as well)